lib/hawkular/operations/operations_api.rb

```diff
@@ -107,12 +123,13 @@ def initialize(args)
         end
       end
 
-      sleep args[:wait_time]
+      Timeout.timeout(@wait_time) { condition.wait_while { !@ws.open? } }
+      @connecting = false
```

Comment:
if something fails between the part where connecting is set to true, to the part where connecting to false, then won't the connecting remain true when it shouldn't?